lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1535,7 +1535,18 @@ namespace {
         if (Name.str() == "id" && Impl.SwiftContext.LangOpts.EnableIdAsAny) {
           Impl.SpecialTypedefNames[Decl->getCanonicalDecl()] =
               MappedTypeNameKind::DoNothing;
-          return Impl.SwiftContext.getAnyDecl();
+
+          auto DC = Impl.importDeclContextOf(Decl, importedName.EffectiveContext);
+          if (!DC) return nullptr;
+
+          // create arbitrary Decl for the implicit Any type
+          return Impl.createDeclWithClangNode<TypeAliasDecl>(
```

Comment:
I don't think this is a good idea; it puts "id" back into the public interface. @jckarter, why is this necessary at all? Doesn't `importType` already handle this?